dashboard-frontend/src/pages/AccountsPage/AccountList.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useState } from 'react';
+import { AccountResource } from '../../types/account';
+import PrimaryButton from '../../components/PrimaryButton';
+import InputTypeformKey from './InputTypeformKey';
+import InputFlyAPIKey from './InputFlyAPIKey';
+import useCreateAccount from './useCreateAccount';
+
+const AccountList: React.FC<any> = ({
+  accounts,
+  clearCreateAccounts,
+}: {
+  accounts: AccountResource[];
+  clearCreateAccounts: () => void;
+}) => {
+  return (
+    <ListLayout>
+      {accounts.map((account, index) => (
+        <AccountListItem
+          key={account.authType}
+          account={account}
+          index={index}
+          credentials={account.connectedAccount?.credentials}
+          clearCreateAccounts={clearCreateAccounts}
+        />
+      ))}
+    </ListLayout>
+  );
+};
+
+const AccountListItem: React.FC<any> = ({
+  account,
+  index,
+  credentials,
+  clearCreateAccounts,
+}: {
+  account: AccountResource;
+  index: number;
+  credentials: any;
+  clearCreateAccounts: () => void;
+}) => {
+  const { isCreating, errorOnCreate, createAccount } = useCreateAccount();
+  const [credential, setCredential] = useState(credentials);
+
+  // used to handle the state of the credentials attached
+  // to the account
+  function handleCredentialChange(e: any): void {
+    setCredential({
+      ...credential,
+      [e.target.name]: e.target.value,
+    });
+  }
+
+  // a bit of duck typing to return the correct input box
+  // TODO this seems complicated, we shouldnt need it
+  const inputType: any = {
+    typeform: (
+      <InputTypeformKey
+        error={errorOnCreate}
+        handleCredentialChange={handleCredentialChange}
+        entity={account.authType}
+        index={index}
+        typeformkey={credential.key}
+      />
+    ),
+    fly: (
+      <InputFlyAPIKey
+        error={errorOnCreate}
+        handleCredentialChange={handleCredentialChange}
+        index={index}
+        entity={account.authType}
+        flyApiKey={credential.api_key}
+      />
+    ),
+  };
+
+  const validateCredentials = JSON.parse(
+    JSON.stringify(credential),
```

Comment:
Unless we want to intentionally return "undefined", we can just say does a value exist? if true give me the value otherwise give me null. 

The shorthand way of writing this would be `value ?? value`

Reply:
This is left over :sweat_smile: I didnt write this, but I can update that